src/main/java/services/DatafeedEventsServiceV2.java

```diff
@@ -84,10 +83,16 @@ public void stopDatafeedService() {
         this.started.set(false);
         if (threadPool != null) {
             threadPool.shutdown();
-            if (!threadPool.awaitTermination(AWAIT_TERMINATION, TimeUnit.MILLISECONDS)) {
+            try {
+                if (!threadPool.awaitTermination(AWAIT_TERMINATION, TimeUnit.MILLISECONDS)) {
+                    threadPool.shutdownNow();
+                    if (!threadPool.awaitTermination(AWAIT_TERMINATION, TimeUnit.MILLISECONDS))
+                        System.err.println("Pool did not terminate");
```

Comment:
We always want to use a logger, never system err (you don't know where it ends up whereas we configure logger, have levels and write them to specific location)

Reply:
oops! My bad.